src/popup.tsx

```diff
@@ -1,90 +1,16 @@
-				for (const provider of Object.values(providers)) {
-					const logoURL = chrome.runtime.getURL(`assets/${provider.name}.svg`);
-					const new_form = document.createElement('form');
-					new_form.setAttribute('action', provider.signinUrl);
-					new_form.setAttribute('target', '_blank');
-					new_form.setAttribute('method', 'POST');
-					new_form.style.width = '100%';
-					new_form.innerHTML = `
-						<input class="csrfToken" type="hidden" name="csrfToken" />
-						<button type="submit" class="button">
-							<img src="${logoURL}" alt="${provider.name}" />
-							<span>Log in with ${provider.name}</span>
-						</button>`;
-					loginDiv?.appendChild(new_form);
-				}
-			});
-
-			fetch(`${websiteUrl}/api/auth/csrf`, { cache: 'no-store' }).then(async (res) => {
-				const json = await res.json();
-				const csrf = json.csrfToken;
-				const csrf_input_elements = document.getElementsByClassName('csrfToken');
-				for (const element of csrf_input_elements) {
-					(element as HTMLInputElement).value = csrf;
-				}
-			});
-
-			fetch(`${websiteUrl}/api/auth/session`, { cache: 'no-store' }).then(async (res) => {
-				const json = await res.json();
-				const loadingDiv = document.querySelector('#loading-div') as HTMLDivElement;
-				const sessionDiv = document.querySelector('#session-div') as HTMLDivElement;
-				const loginDiv = document.querySelector('#login-div') as HTMLDivElement;
-
-				if (loadingDiv && sessionDiv && loginDiv) {
-					loadingDiv.style.display = 'none';
-
-					if (json.user) {
-						// User is logged in
-						sessionDiv.style.display = 'flex';
-						const signoutForm = document.getElementById('signout-form') as HTMLFormElement;
-						signoutForm.setAttribute('action', `${websiteUrl}/api/auth/signout`);
-
-						const { user } = json;
-						const sessionImage = document.querySelector('#session-image') as HTMLImageElement;
-						const sessionName = document.querySelector('#session-name') as HTMLHeadingElement;
-						const sessionEmail = document.querySelector('#session-email') as HTMLSpanElement;
-
-						sessionImage.src = user.image;
-						sessionName.innerHTML = user.name;
-						sessionEmail.innerHTML = user.email;
-
-						chrome.storage.sync.set({
-							userId: user.id,
-							userName: user.name,
-							userImage: user.image
-						}).then(() => {
-							console.debug(`[popup] userId has been set to ${user.id}`);
-						}).catch(err => {
-							console.error(`[popup] Sync storage could not be set. userId: ${user.id}`, err);
-						});
-					} else {
-						// No session means user not logged in
-						loginDiv.style.display = 'flex';
-					}
-				}
-			});
-		});
-	}, []);
+import LoadingDiv from './LoadingDiv';
+import LoginDiv from './LoginDiv';
+import SessionDiv from './SessionDiv';
 
+import useFetchPopupData from './fetchPopupEffects';
+import useVersion from '~VersionUpdate';
 
+const IndexPopup: React.FC<{}> = () => {
+	const version = useVersion(chrome);
+	useFetchPopupData(chrome);
```

Comment:
Oh my my! This is not modularization at all!! You have just broken the code into different files! There is no way anyone can use these components in any modular way!

A single component must contain all its logic within itself. You can't just call `useFetchPopupData` at a single place and yet call all the other pieces as independent components.